src/python/pants/backend/scala/dependency_inference/rules.py

```diff
@@ -83,11 +99,76 @@ async def infer_scala_dependencies_via_source_analysis(
     return InferredDependencies(dependencies)
 
 
+class InjectScalaLibraryDependencyRequest(InjectDependenciesRequest):
+    inject_for = ScalaDependenciesField
+
+
+@dataclass(frozen=True)
+class ResolveScalaLibraryTargetForResolveRequest:
+    resolve_name: str
+
+
+@dataclass(frozen=True)
+class ResolvedScalaLibraryTargetForResolve:
```

Comment:
We've tried to avoid subtle naming differences like `Resolved` vs `Resovle`. I recommend `ResolveScalaLibraryTargetForResolve` and `ResolveScalaLibraryTargetForResolveResult`

Also it's confusing that resolve is used twice. Maybe `ScalaLibForResolve` and `ScalaLibForResolveRequest`?

Reply:
Agreed, double resolve in the name is confusing. I like your suggest although library is a bit confusing too. Will use `ScalaRuntimeForResolveRequest` and `ScalaRuntimeForResolve`.